### PR TITLE
Magento 2 4 2 release notes --update 01/26

### PR DESCRIPTION
**Pull request by jfrontain**
## Purpose of this pull request

This pull request (PR) adds additional fix and preliminary known issue descriptions to the Magento 2.4.2 release notes. This is a preliminary set of notes and will change before GA.

## Affected DevDocs pages

https://devdocs.magento.com/guides/v2.4/release-notes/bk-release-notes.html

whatsnew
Added preliminary version of the Magento 2.4.2 release notes.

### CODE REVIEW

File: src/guides/v2.4/release-notes/commerce-2-4-2.md [hunk 1]
```diff
@@ -1419,12 +2031,30 @@ We have fixed hundreds of issues in the Magento 2.4.2 core code.
...
+
+### B2B known issues
+
+**Issue**: B2B buyers can use online payment methods to bypass the usual purchase order flow. This scenario can occur if the buyer can reduce their entire checkout total to a 0 — for example, by a promo code or gift card —  and subsequently remove the code or gift card. Even under those conditions, Magento still places the order for the correct amount based on the prices of the items in their assigned catalog. <!--- B2B-1603-->
+
+**Issue**: Buyers are redirected to the shopping cart when trying to place an order from a purchase order using PayPal Express Checkout when **In-Context Mode** is disabled. <!--- B2B-1604-->
+
+**Issue**: Magento randomly When attempting to create a purchase order and navigating to the initial checkout page, a bad request can occasionally be observed in the browser console when a buyer creates a purchase order and navigates to the checkout page. The buyer can still place the purchase order. Magento displays this error when a buyer has previously created a different purchase order with an online payment method before navigating to checkout page without completing the purchase. **Workaround**: None. <!--- B2B-1605-->
```
Comment (dobooth) [inline src/guides/v2.4/release-notes/commerce-2-4-2.md:2056]: Something is amiss at the start of this note.

**Review by dobooth — APPROVED** [review]

File: src/guides/v2.4/release-notes/commerce-2-4-2.md [hunk 2]
```diff
@@ -1419,12 +2031,30 @@ We have fixed hundreds of issues in the Magento 2.4.2 core code.
...
+
+## Known issues
 
 **Issue**: The `deleteRequisitionListOutput` query returns details about the deleted requisition list instead of the remaining requisition lists. <!--- MC-39894-->
 
+**Issue**: Shoppers cannot add a configurable product to their cart from a non-default store view in a multi-store deployment. Magento displays this error: `Could not add item to cart. Please check required options and try again`. **Workaround**: None. [GitHub-31660](https://github.com/magento/magento2/issues/31660)<!--- PWA-1298-->
+
+**Issue**: Merchants must activate the Fastly Force TLS functionality from the Admin to enable the global HTTP to HTTPS redirect for all store pages. See the [Redirect HTTP to HTTPS for all pages on Cloud (Force TLS)](https://support.magento.com/hc/en-us/articles/360006296953) Knowledge Base article. <!--- MC-39988-->
+
+### B2B known issues
```
Comment (joanhe) [inline src/guides/v2.4/release-notes/commerce-2-4-2.md:2050]: Why is B2B known issues in src/guides/v2.4/release-notes/b2b-release-notes.md?

File: src/guides/v2.4/release-notes/commerce-2-4-2.md [hunk 3]
```diff
@@ -1419,12 +2031,30 @@ We have fixed hundreds of issues in the Magento 2.4.2 core code.
...
+
+**Issue**: B2B buyers can use online payment methods to bypass the usual purchase order flow. This scenario can occur if the buyer can reduce their entire checkout total to a 0 — for example, by a promo code or gift card —  and subsequently remove the code or gift card. Even under those conditions, Magento still places the order for the correct amount based on the prices of the items in their assigned catalog. <!--- B2B-1603-->
```
Comment (joanhe) [inline src/guides/v2.4/release-notes/commerce-2-4-2.md:2052]: I suggest to add disabling gift card, coupon code as work around of this issue when online payment methods are enabled for purchase order approval.

File: src/guides/v2.4/release-notes/commerce-2-4-2.md (hunk 1 above)
Comment (jfrontain) [reply src/guides/v2.4/release-notes/commerce-2-4-2.md]: Thanks for catching that ugly copy-and-paste. Corrected.

File: src/guides/v2.4/release-notes/commerce-2-4-2.md (hunk 2 above)
Comment (jfrontain) [reply src/guides/v2.4/release-notes/commerce-2-4-2.md:2050]: good point, Joan -- I've added them to the B2B notes. Thanks.

File: src/guides/v2.4/release-notes/commerce-2-4-2.md (hunk 3 above)
Comment (jfrontain) [reply src/guides/v2.4/release-notes/commerce-2-4-2.md]: Added. Thank you, Joan -- really helpful information.

**Comment by dobooth** [comment]
running tests

File: src/guides/v2.4/release-notes/open-source-2-4-2.md [hunk 4]
```diff
@@ -1098,6 +1614,18 @@ We have fixed hundreds of issues in the Magento 2.4.2 core code.
...
+
+*  Added collapsible navigation functionality to the Blank theme in mobile view. [GitHub-30237](https://github.com/magento/magento2/issues/30237)
+
+<!-- ENGCOM-7803 -->
+
+*  Web font loading for themes has been optimized.  [GitHub-29526](https://github.com/magento/magento2/issues/29526)
+
+<!-- ENGCOM-8419 -->
+
+*  Deployment of specific themes for multiple locales has been optimized, which improves deployment that use a single theme with different locales. [GitHub-30184](https://github.com/magento/magento2/issues/30184)
```
Comment (hostep) [inline src/guides/v2.4/release-notes/open-source-2-4-2.md:1627]: The description of the issue no longer reflects what actually was approved.
So here is what I suggest should be used as a more accurate description:

> A new `--no-parent` flag was added to the `bin/magento setup:static-content:deploy` command which prevents the parent(s) of a theme to be compiled (if any). Often this was unnecessary and with this new flag you can significantly improve the performance of the static content deployment process. However, be aware that this new flag does not work when using the `compact` strategy.

File: src/guides/v2.4/release-notes/open-source-2-4-2.md [hunk 5]
```diff
@@ -1218,9 +1786,19 @@ We have fixed hundreds of issues in the Magento 2.4.2 core code.
...
+<!-- ENGCOM-7779 -->
+
+*  Implemented `ActionInterface` for \Magento\Wishlist\Controller\Shared\Allcart. [GitHub-29537](https://github.com/magento/magento2/issues/29537)
+
+## Known issues
+
+**Issue**: The `deleteRequisitionListOutput` query returns details about the deleted requisition list instead of the remaining requisition lists. <!--- MC-39894-->
 
-**Issue**: Discounts for a specific payment method persist during checkout for a purchase order even when the buyer changes payment method during final checkout. As a result, customers may receive a discount that they are not entitled to. This occurs because a cart rule for the original payment method is still applied despite the change in payment method. **Workaround**: None. <!-- KB-827 -->
+**Issue**: Shoppers cannot add a configurable product to their cart from a non-default store view in a multi-store deployment. Magento displays this error: `Could not add item to cart. Please check required options and try again`. **Workaround**: None. [GitHub-31660](https://github.com/magento/magento2/issues/31660)<!--- PWA-1298-->
```
Comment (hostep) [inline src/guides/v2.4/release-notes/open-source-2-4-2.md:1801]: Is this issue only related to the PWA storefront, or also to the default Magento storefront? If it's only PWA, you'd need to explicitly mention this as to not scare users from upgrading with this possibly horrible bug.

**Review by hostep — COMMENTED** [review]
@jfrontain: I've left some small comments behind here and there :)

File: src/guides/v2.4/release-notes/open-source-2-4-2.md (hunk 4 above)
Comment (jfrontain) [reply src/guides/v2.4/release-notes/open-source-2-4-2.md:1627]: thanks for the update -- edited and added to the notes.

File: src/guides/v2.4/release-notes/open-source-2-4-2.md [hunk 6]
```diff
@@ -586,6 +754,128 @@ We have fixed hundreds of issues in the Magento 2.4.2 core code.
...
+
+*  Customer data is now retrieved as expected when you execute a custom login controller. Previously,reloading all `customer-data` sections by wildcard (*) caused requests to `customer/section/load` to return a 400. [GitHub-28428](https://github.com/magento/magento2/issues/28428)
+
+<!-- ENGCOM-8022 -->
+
+*  The framework/database/select `where` functions now handle the `type` correctly. [GitHub-29590](https://github.com/magento/magento2/issues/29590)
+
+<!-- ENGCOM-8260 -->
+
+*  Issues with `array_merge` in loops has been fixed throughout the codebase, which improves static content deployment. [GitHub-30183](https://github.com/magento/magento2/issues/30183)
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.4/release-notes/open-source-2-4-2.md:839]: thanks -- this was one of the issues I was going to ask you about -- have edited slightly and added to the notes.

File: src/guides/v2.4/release-notes/open-source-2-4-2.md [hunk 7]
```diff
@@ -536,12 +688,28 @@ We have fixed hundreds of issues in the Magento 2.4.2 core code.
...
+
+*  Images on a store’s home page are now rendered successfully. Previously, the page source HTML was corrupted because the three consecutive slashes in the image’s base64 code was misinterpreted as a comment.
+
 <!--- MC-37369-->
 
 *  DatePicker now works correctly when filtering orders on the Admin Orders list in stores using the Arabic locale (`ar_SA - Saudi Arabia`).
 
+<!--- MC-38344-->
+
+*  DatePicker now displays the correct date after a merchant saves a cart price rule in a deployment that implements a Dutch (Netherlands) locale. [GitHub-30382](https://github.com/magento/magento2/issues/30382)
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.4/release-notes/open-source-2-4-2.md:711]: Will need to follow up after we merge this PR. I'll update as needed in the next version (and as you know, there's always another version...)

File: src/guides/v2.4/release-notes/open-source-2-4-2.md (hunk 5 above)
Comment (jfrontain) [reply src/guides/v2.4/release-notes/open-source-2-4-2.md:1801]: @hostep, I am going to remove this known issue description from this PR and add it to the next PR later this week when I can get clarification. (I have other known issue descriptions to add, too.) thanks.

**Comment by jfrontain** [comment]
running tests

File: src/guides/v2.4/release-notes/commerce-2-4-2.md [hunk 8]
```diff
@@ -153,7 +151,7 @@ This release includes enhancements to Page Builder content migration and Page Bu
 
 *  The new mobile [viewport switcher](https://docs-beta.magento.com/user-guide/cms/page-builder-workspace.html#viewports) and viewport scope for form field values lets users and developers perform these actions:
 
-   *  View content on different viewports when authoring.
+   *  View content on different view ports when authoring.
```
Comment (danmooney2) [inline src/guides/v2.4/release-notes/commerce-2-4-2.md:154]: Original wording is correct

File: src/guides/v2.4/release-notes/b2b-release-notes.md [hunk 9]
```diff
@@ -55,6 +55,20 @@ These release notes can include:
...
 
+-  {:.fix} The `credit_history` query now returns the specified company’s credit history for both the originally allocated amount and the purchased amount. Previously, this query returned an error. [GitHub-29990](https://github.com/magento/magento2/issues/29990)
+
+-  {:.fix} The **Company** and  **Job Title** fields on the Edit Account Information page are no longer editable. [GitHub-312](https://github.com/magento/partners-magento2b2b/issues/312)
+
+### B2B known issues
+
+**Issue**: B2B buyers can use online payment methods to bypass the usual purchase order flow. This scenario can occur if the buyer can reduce their entire checkout total to a 0 — for example, by a promo code or gift card —  and subsequently remove the code or gift card. Even under those conditions, Magento still places the order for the correct amount based on the prices of the items in their assigned catalog. **Workaround**: Disable gift cards and coupon codes when online payment methods are enabled for purchase order approval. <!--- B2B-1603-->
```
Comment (danmooney2) [inline src/guides/v2.4/release-notes/b2b-release-notes.md:64]: `to a 0` sounds a little funky.

File: src/guides/v2.4/release-notes/commerce-2-4-2.md (hunk 8 above)
Comment (danmooney2) [reply src/guides/v2.4/release-notes/commerce-2-4-2.md:154]: I see there's a lot of separation of commonly accepted compound words in this PR.  I don't think it's a big deal for search engines.

**Pull request merged**